pennylane/debugging/snapshot.py

```diff
@@ -172,6 +172,66 @@ def circuit():
       '10': tensor(48, requires_grad=True)},
      'execution_results': tensor(-0.1, requires_grad=True)}
 
+    **Integration with Mid Circuit Measurements:**
+
+    Mid circuit measurements can be handled either by executing the circuit ``shots`` times (``mcm_method="one-shot"``),
+    or constructing and executing a tree of all possible branches (``mcm_method="tree-traversal"``).
+
+    .. code-block:: python
+
+        @qml.qnode(qml.device('default.qubit'))
+        def c(y):
+            qml.H(0)
+            qml.Snapshot(measurement=qml.probs(0))
+            m0 = qml.measure(0)
+            qml.cond(m0, qml.RX)(0.5, 0)
+            qml.Snapshot(measurement=qml.probs(0))
+            qml.RY(y, 0)
+            qml.measure(0)
+            qml.Snapshot(measurement=qml.probs(0))
+            return qml.probs(0)
+
+    If we use ``"one-shot"`` together with three shots, you can see that each snapshot has three
+    probabilities in a list.  Each entry corresponds to a different execution of the circuit. Note that
+    postselection will not be applied on snapshots with ``"one-shot"``.
+
+    >>> one_shot = qml.set_shots(c, 3).update(mcm_method="one-shot")
+    >>> qml.snapshots(one_shot)(1.0)
+    {0: [array([0., 1.]), array([0., 1.]), array([1., 0.])],
+    1: [array([1., 0.]), array([1., 0.]), array([1., 0.])],
+    2: [array([1., 0.]), array([1., 0.]), array([1., 0.])],
+    'execution_results': array([1., 0.])}
+
+    Note that for a single shot ``"one-shot"`` execution, the results will not be in a list.
+
+    >>> qml.snapshots(qml.set_shots(one_shot, 1))(1.0)
+    {0: array([0., 1.]),
+    1: array([0., 1.]),
+    2: array([0., 1.]),
+    'execution_results': array([0., 1.])}
```

Comment:
I think it's worth keeping the example slightly more self contained:
```suggestion
    Note that for a single shot ``"one-shot"`` execution, the results will not be in a list.

    >>> one_shot = qml.set_shots(c, 1).update(mcm_method="one-shot")
    >>> qml.snapshots(one_shot)(1.0)
    {0: array([0., 1.]),
    1: array([0., 1.]),
    2: array([0., 1.]),
    'execution_results': array([0., 1.])}
```